### PR TITLE
Adds Poisson 2D MaskFunc

### PR DESCRIPTION
**Pull request by jerke123**
Added Poisson2d as a mask option in subsample.py

### CODE REVIEW

File: mridc/data/subsample.py [hunk 1]
```diff
@@ -407,6 +408,256 @@ def gaussian_kernel(self):
...
+        mask = np.logical_or(pattern1, pattern2)
+
+        if half_scan_percentage != 0:
+            mask[: int(np.round(mask.shape[0] * half_scan_percentage)), :] = 0.0
+
+        return (torch.from_numpy(mask.astype(np.float32)).unsqueeze(0).unsqueeze(-1), acceleration)
+
+    def poisson_disc2d(self):
+        k = 10
+        centerscale = self.scale
```
Comment (wdika) [inline mridc/data/subsample.py]: Unused variable, please remove.

File: mridc/data/subsample.py [hunk 2]
```diff
@@ -407,6 +408,256 @@ def gaussian_kernel(self):
...
+            yvals = np.arange(coords[1] - rx, coords[1] + rx)
+
+            xvals = xvals[(xvals >= 0) & (xvals <= width)]
+            yvals = yvals[(yvals >= 0) & (yvals <= height)]
+
+            dist = lambda x, y: np.sqrt((coords[0] - x) ** 2 + (coords[1] - y) ** 2) < rx
+            xx, yy = np.meshgrid(xvals, yvals, sparse=False)
+            pts = np.vstack((xx.ravel(), yy.ravel())).T
+            pts = pts[dist(pts[:, 0], pts[:, 1])]
+            [cells[get_cell_coords(pt)].append(idx) for pt in pts]
```
Comment (wdika) [inline mridc/data/subsample.py]: Expression not assigned. Is this returned?

File: mridc/data/subsample.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 from typing import Optional, Sequence, Tuple, Union
 
 import numpy as np
+from numpy import linalg as LA
 import torch
 from numpy import ndarray
```
Comment (wdika) [inline mridc/data/subsample.py]: Please group numpy ndarray and linalg imports.

File: mridc/data/subsample.py [hunk 4]
```diff
@@ -407,6 +408,256 @@ def gaussian_kernel(self):
...
+            """Calculate r for the given coordinates."""
+            return ((LA.norm(np.asarray(coords) - center) / centernorm) * 240 + 50) / self.r
+
+        def get_cell_coords(pt):
+            """Get the coordinates of the cell that pt = (x,y) falls in."""
+            return int(pt[0] // a), int(pt[1] // a)
+
+        def mark_neighbours(idx):
+            """
+            Add sample index to the cells within r(point) range of the point.
```
Comment (wdika) [inline mridc/data/subsample.py]: One-line docstring. Please fit it on one line as above.

File: mridc/data/subsample.py [hunk 5]
```diff
@@ -407,6 +408,256 @@ def gaussian_kernel(self):
...
+        pattern1 = self.poisson_disc2d()
+        pattern2 = self.centered_circle()
+        mask = np.logical_or(pattern1, pattern2)
+
+        if half_scan_percentage != 0:
+            mask[: int(np.round(mask.shape[0] * half_scan_percentage)), :] = 0.0
+
+        return (torch.from_numpy(mask.astype(np.float32)).unsqueeze(0).unsqueeze(-1), acceleration)
+
+    def poisson_disc2d(self):
```
Comment (wdika) [inline mridc/data/subsample.py:516]: Please add docstring to this func.

File: tests/test_subsample.py [hunk 6]
```diff
@@ -114,9 +115,11 @@ def test_create_mask_for_gaussian1d_type(
     Returns:
         None
     """
+    print("1")
     mask_func = create_mask_for_mask_type(mask_type, center_fractions, accelerations)
-
+    print("1")
     mask, acc = mask_func(x, seed, half_scan_percentage, scale)
+    print("1")
```
Comment (wdika) [inline tests/test_subsample.py]: Please remove those print statements.

**Review by wdika — CHANGES_REQUESTED** [review]
Hi @jerke123, thanks for your PR. Can you please make a few fixes according to my comments? So your code matches with the repo's code quality standards.

**Review by wdika — APPROVED** [review]
